### ISSUE TITLE
More frequent (or configurable) update checks

### ISSUE THREAD

**Issue by northys** [safing/portmaster#578]
**What would you like to add or change?**:

I noticed that I'm running 0.8.5 but yesterday portmaster 0.8.6 was released and 30 minutes ago 0.8.7 was released. It would be nice if portmaster could check for updates more frequent.

This could be triggered on system wake up maybe? Or configurable... Or just more frequently.

Serving that little static json using nginx is cached in ram perfently and there is no problem to serve thousands of those update jsons per second on 5$ VPS so I think server performance ("DDOS") is not the issue :)

**Why do you and others need this?**:

I can't wait to have new features and bugfixes!

**Comment by dhaavi** [comment]
A Github "release" is primarily a tag and does not necessarily mean that the version is actually released yet.

I prepared everything for a release yesterday, so some dependencies were released on Github yesterday or even earlier. Today, I found a critical bug in v0.8.6 during staging, so that version was never released. I fixed the issue in v0.8.7, which also went into staging first and later into beta.

I think hourly checks for updates is more than enough. ;)

TL;DR: Tagging on Github only means that a version is about to go into staging, and will be released to the beta release channel in the near unspecified future, IF it passes staging.

**Comment by northys** [comment]
Why I got the 0.8.7 update then when I trigered the update check manually? My computer was running for a few hours already before I noticed you tagged new version. When I triggered the update it said 0.8.7 is ready and portmaster can be restarted to it.

I have set the most edge updates channel available so I think I have all updates sooner.

**Comment by northys** [comment]
okay, you are right. it checks for update every hour but it did not show the "upgraded, pls restart me" notification.

```
bře 23 01:49:22 northys-fedora portmaster-start[898]: 220323 01:49:22.091 r/updating:102 ▶ INFO 742 updates: updated index stable.json with 29 entries
bře 23 02:49:22 northys-fedora portmaster-start[898]: 220323 02:49:22.214 r/updating:102 ▶ INFO 385 updates: updated index stable.json with 29 entries
bře 23 03:49:23 northys-fedora portmaster-start[898]: 220323 03:49:23.015 r/updating:102 ▶ INFO 068 updates: updated index stable.json with 29 entries
bře 23 04:49:23 northys-fedora portmaster-start[898]: 220323 04:49:23.270 r/updating:102 ▶ INFO 054 updates: updated index stable.json with 29 entries
bře 23 15:32:28 northys-fedora portmaster-start[898]: 220323 15:32:28.677 r/updating:102 ▶ INFO 943 updates: updated index stable.json with 29 entries
bře 23 16:32:43 northys-fedora portmaster-start[898]: 220323 16:32:43.410 r/updating:102 ▶ INFO 774 updates: updated index stable.json with 29 entries
bře 23 17:32:43 northys-fedora portmaster-start[898]: 220323 17:32:43.747 r/updating:102 ▶ INFO 194 updates: updated index stable.json with 29 entries
bře 23 18:32:44 northys-fedora portmaster-start[898]: 220323 18:32:44.145 r/updating:102 ▶ INFO 080 updates: updated index stable.json with 29 entries
bře 23 19:32:44 northys-fedora portmaster-start[898]: 220323 19:32:44.850 r/updating:102 ▶ INFO 020 updates: updated index stable.json with 29 entries
bře 23 20:32:45 northys-fedora portmaster-start[898]: 220323 20:32:45.281 r/updating:102 ▶ INFO 143 updates: updated index stable.json with 29 entries
bře 23 21:32:45 northys-fedora portmaster-start[898]: 220323 21:32:45.838 r/updating:102 ▶ INFO 183 updates: updated index stable.json with 29 entries
bře 23 22:32:46 northys-fedora portmaster-start[898]: 220323 22:32:46.957 r/updating:102 ▶ INFO 100 updates: updated index stable.json with 29 entries
bře 23 23:32:48 northys-fedora portmaster-start[898]: 220323 23:32:48.361 r/updating:102 ▶ INFO 218 updates: updated index stable.json with 29 entries
bře 24 00:32:48 northys-fedora portmaster-start[898]: 220324 00:32:48.628 r/updating:102 ▶ INFO 998 updates: updated index stable.json with 29 entries
bře 24 01:32:49 northys-fedora portmaster-start[898]: 220324 01:32:49.623 r/updating:102 ▶ INFO 425 updates: updated index stable.json with 29 entries
bře 24 02:32:51 northys-fedora portmaster-start[898]: 220324 02:32:51.343 r/updating:102 ▶ INFO 833 updates: updated index stable.json with 29 entries
bře 24 15:22:10 northys-fedora portmaster-start[898]: 220324 15:22:10.317 r/updating:102 ▶ INFO 421 updates: updated index stable.json with 35 entries
bře 24 16:20:45 northys-fedora portmaster-start[898]: 220324 16:20:45.769 r/updating:102 ▶ INFO 071 updates: updated index stable.json with 35 entries
bře 24 17:20:52 northys-fedora portmaster-start[898]: 220324 17:20:52.883 r/updating:102 ▶ INFO 209 updates: updated index stable.json with 35 entries
bře 24 18:20:53 northys-fedora portmaster-start[898]: 220324 18:20:53.023 r/updating:102 ▶ INFO 337 updates: updated index stable.json with 35 entries
bře 24 19:32:33 northys-fedora portmaster-start[898]: 220324 19:32:33.910 r/updating:102 ▶ INFO 667 updates: updated index stable.json with 35 entries
bře 24 19:35:40 northys-fedora portmaster-start[898]: 220324 19:35:40.249 r/updating:103 ▶ INFO 064 updates: updated index stable.json with 35 entries
bře 24 19:37:13 northys-fedora portmaster-start[898]: 220324 19:37:13.392 r/updating:103 ▶ INFO 193 updates: updated index stable.json with 35 entries
bře 24 19:38:21 northys-fedora portmaster-start[898]: 220324 19:38:21.581 r/updating:103 ▶ INFO 343 updates: updated index stable.json with 35 entries
bře 24 19:38:30 northys-fedora portmaster-start[898]: 220324 19:38:30.197 r/updating:103 ▶ INFO 517 updates: updated index stable.json with 35 entries
bře 24 19:39:02 northys-fedora portmaster-start[898]: 220324 19:39:02.479 r/updating:103 ▶ INFO 545 updates: updated index stable.json with 35 entries
bře 24 20:39:02 northys-fedora portmaster-start[898]: 220324 20:39:02.620 r/updating:103 ▶ INFO 804 updates: updated index stable.json with 35 entries
bře 24 21:39:05 northys-fedora portmaster-start[898]: 220324 21:39:05.102 r/updating:103 ▶ INFO 091 updates: updated index stable.json with 35 entries
bře 24 22:39:06 northys-fedora portmaster-start[898]: 220324 22:39:06.465 r/updating:103 ▶ INFO 855 updates: updated index stable.json with 35 entries
bře 24 23:39:07 northys-fedora portmaster-start[898]: 220324 23:39:07.392 r/updating:103 ▶ INFO 623 updates: updated index stable.json with 35 entries
bře 25 00:39:07 northys-fedora portmaster-start[898]: 220325 00:39:07.562 r/updating:103 ▶ INFO 950 updates: updated index stable.json with 35 entries
```

**Comment by northys** [comment]
> Today, I found a critical bug in v0.8.6 during staging, so that version was never released. I fixed the issue in v0.8.7, which also went into staging first and later into beta.

Okay after I read that again I got it. I expected update to 0.8.7 which was released only for 30 minutes and manual update updated it so there is chance I would get that notification in another 30 minutes. Closing.
